internal/cache/clockpro.go

```diff
@@ -299,8 +299,8 @@ func (c *shard) metaAdd(key key, e *entry) bool {
 		c.handHot.link(e)
 	}
 
-	if c.handCold == c.handHot {
-		c.handCold = c.handCold.prev()
+	if c.sizeCold == 0 {
```

Comment:
If added entry 'e' is a fresh cold type entry but the handHot pointed entry is also cold type entry, then handCold should not point the 'e', because it has a smaller recency than handHot pointed entry. Additionally, always moving handCold to handHot.prev() when the sizeCold == 0 is not harmful because it is the best place to find the next added cold entry quickly.